java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1030,9 +848,9 @@ public ForStatementTreeImpl newStandardForStatement(
     StatementExpressionListTreeImpl forInit2 = forInit.isPresent() ? forInit.get() : new StatementExpressionListTreeImpl(ImmutableList.<StatementTree>of());
     StatementExpressionListTreeImpl forUpdate2 = forUpdate.isPresent() ? forUpdate.get() : new StatementExpressionListTreeImpl(ImmutableList.<StatementTree>of());
 
-    InternalSyntaxToken forKeyword = (InternalSyntaxToken) forTokenAstNode;
+    InternalSyntaxToken forKeyword = (InternalSyntaxToken) forTokenKeyword;
 
-    ForStatementTreeImpl result = new ForStatementTreeImpl(
+    return new ForStatementTreeImpl(
       forKeyword,
       openParenToken,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)